backend/infrahub/core/migrations/query/node_duplicate.py

```
@@ -38,11 +38,23 @@ def __init__(
 
     def render_match(self) -> str:
         labels_str = ":".join(self.previous_node.labels)
-        query = f"""
+        query = """
         // Find all the active nodes
-        MATCH (node:{labels_str})
+        MATCH (node:%(labels_str)s)
         WITH DISTINCT node
-        """
+        CALL (node) {
+            WITH labels(node) AS node_labels
+            UNWIND node_labels AS n_label
+            ORDER BY n_label ASC
+            WITH collect(n_label) AS sorted_labels
+
+            RETURN (
+                node.kind = $new_node.kind AND
+                sorted_labels = $new_sorted_labels
+            ) AS already_migrated
+        }
+        WITH node WHERE already_migrated = FALSE
+        """ % {"labels_str": labels_str}
```

Is it possible to add a docstring to `class NodeDuplicateQuery(Query)` to indicate what's it's doing? It took me some time to re-understand that we are copying all nodes from a given kind to another.

Also a comment around above block would be helpful to indicate why we need this block, ie because this migration might have silently failed for unkown reasons resulting in nodes not having being migrated

---

I'll add a comment or two, but the changes are really just to make the migration idempotent so that we don't try to migrate nodes unnecessarily. for example, if the same schema changes are made on `main` and a branch before that branch is merged into `main`, then we could skip basically all of this migration logic during the merge process